models/user.js

```diff
@@ -18,11 +20,46 @@ const userSchema = new mongoose.Schema({
     required: true,
     validate: {
       validator(v) {
-        return /^(ftp|http|https):\/\/[^ "]+$/.test(v);
+        // return /^(ftp|http|https):\/\/[^ "]+$/.test(v);
+        return isURL(v);
       },
       message: (props) => `${props.value} is not a valid link!`,
     },
   },
+  email: {
+    type: String,
+    required: true,
+    unique: true,
+    validate: {
+      validator(v) {
+        return isEmail(v);
+      },
+      message: (props) => `${props.value} is not a valid email!`,
+    },
+  },
+  password: {
+    type: String,
+    required: true,
+    minlength: 8,
+    select: false,
+  },
 });
 
+// eslint-disable-next-line func-names
+userSchema.statics.findUserByCredentials = function (email, password) {
+  return this.findOne({ email }).select('+password')
+    .then((user) => {
+      if (!user) {
+        return Promise.reject(new Error('Неправильная почта или пароль!'));
+      }
+      return bcrypt.compare(password, user.password)
+        .then((matched) => {
+          if (!matched) {
+            return Promise.reject(new Error('Неправильная почта или пароль!'));
```

Comment:
Можно лучше: перенести класс ошибки из контроллера.
```suggestion
            return Promise.reject(new UnathorizedError('Неправильная почта или пароль!'));
```